diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 0289a97325d1..cc7790dbb4b9 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -41,6 +41,7 @@ riscv-march-$(CONFIG_ARCH_RV32I)	:= rv32ima
 riscv-march-$(CONFIG_ARCH_RV64I)	:= rv64ima
 riscv-march-$(CONFIG_FPU)		:= $(riscv-march-y)fd
 riscv-march-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-y)c
+riscv-march-y := $(riscv-march-y)_zicsr_zifencei
 KBUILD_CFLAGS += -march=$(subst fd,,$(riscv-march-y))
 KBUILD_AFLAGS += -march=$(riscv-march-y)
 
